mlrun/api/schemas/model_endpoints.py

```diff
@@ -22,73 +28,225 @@ class ModelEndpointSpec(ObjectSpec):
+
+        Parameters for ModelEndpointStatus
+        :param state: The current state of the endpoint
+        :param active: The "activation" status of the endpoint - True for active / False for not active (default True)
+        """
+
+        if project is None and model is None and function is None:
+            raise MLRunInvalidArgumentError(
+                f"All of 'project[={project}]', 'model[={model}]', 'function[={function}]' must be properly initialized"
             )
 
-        endpoint_unique_string = (
-            f"{self.spec.function}_{self.spec.model}_{self.metadata.tag}"
+        return cls(
+            metadata=ModelEndpointMetadata(
+                project=project,
+                tag=tag,
+                labels=labels or {},
+                model_artifact=model_artifact,
+                stream_path=stream_path,
+                feature_stats=feature_stats,
+                feature_names=feature_names,
+                monitor_configuration=monitor_configuration,
+            ),
+            spec=ModelEndpointSpec(
+                model=model, function=function, model_class=model_class,
+            ),
+            status=ModelEndpointStatus(state=state, active=active),
         )
-        md5_str = md5(endpoint_unique_string.encode("utf-8")).hexdigest()
-        return f"{self.metadata.project}.{md5_str}"
 
+    @property
+    def project(self):
+        return self.metadata.project
 
-class Histogram(BaseModel):
-    buckets: List[Tuple[float, float]]
-    count: List[int]
+    @property
+    def function(self):
+        return self.spec.function
+
+    @property
+    def model(self):
+        return self.spec.model
+
+    @property
+    def tag(self):
+        return self.metadata.tag
 
 
 class Metric(BaseModel):
     name: str
-    start_timestamp: str
-    end_timestamp: str
-    headers: List[str]
     values: List[Tuple[str, float]]
-    min: float
-    avg: float
-    max: float
+
+
+class Histogram(BaseModel):
+    buckets: List[float]
+    counts: List[int]
 
 
 class FeatureValues(BaseModel):
     min: float
-    avg: float
+    mean: float
     max: float
     histogram: Histogram
 
+    @classmethod
+    def new(cls, stats: Optional[dict]):
+        if stats:
+            return FeatureValues(
+                min=stats["min"],
+                mean=stats["mean"],
+                max=stats["max"],
+                histogram=Histogram(buckets=stats["hist"][1], counts=stats["hist"][0]),
+            )
+        else:
+            return None
+
 
 class Features(BaseModel):
     name: str
     weight: float
-    expected: FeatureValues
+    expected: Optional[FeatureValues]
     actual: Optional[FeatureValues]
 
+    @classmethod
+    def new(
+        cls,
+        feature_name: str,
+        feature_stats: Optional[dict],
+        current_stats: Optional[dict],
+    ):
+        return cls(
+            name=feature_name,
+            weight=-1.0,
+            expected=FeatureValues.new(feature_stats),
```

Comment:
You could simply do `FeatureValues(**feature_stats)` and you wouldn't need this `new`

Reply:
The `feature_stats` dictionary is not generated by me, and it does not align perfectly with this object